monitoring/dashboard.json

```diff
@@ -0,0 +1,627 @@
+            "type": "timeseries"
+        },
+        {
+            "datasource": null,
+            "description": "",
+            "fieldConfig": {
+                "defaults": {
+                    "color": {
+                        "mode": "palette-classic"
+                    },
+                    "custom": {
+                        "axisLabel": "",
+                        "axisPlacement": "auto",
+                        "barAlignment": 0,
+                        "drawStyle": "line",
+                        "fillOpacity": 0,
+                        "gradientMode": "none",
+                        "hideFrom": {
+                            "legend": false,
+                            "tooltip": false,
+                            "viz": false
+                        },
+                        "lineInterpolation": "linear",
+                        "lineWidth": 1,
+                        "pointSize": 5,
+                        "scaleDistribution": {
+                            "type": "linear"
+                        },
+                        "showPoints": "auto",
+                        "spanNulls": false,
+                        "stacking": {
+                            "group": "A",
+                            "mode": "none"
+                        },
+                        "thresholdsStyle": {
+                            "mode": "off"
+                        }
+                    },
+                    "mappings": [],
+                    "thresholds": {
+                        "mode": "absolute",
+                        "steps": [
+                            {
+                                "color": "green",
+                                "value": null
+                            },
+                            {
+                                "color": "red",
+                                "value": 80
+                            }
+                        ]
+                    },
+                    "unit": "ms"
+                },
+                "overrides": []
+            },
+            "gridPos": {
+                "h": 8,
+                "w": 12,
+                "x": 12,
+                "y": 10
+            },
+            "id": 14,
+            "options": {
+                "legend": {
+                    "calcs": [],
+                    "displayMode": "list",
+                    "placement": "bottom"
+                },
+                "tooltip": {
+                    "mode": "single"
+                }
+            },
+            "targets": [
+                {
+                    "exemplar": true,
+                    "expr": "sum(rate(http_request_duration_seconds_sum{namespace=\"${namespace}\", job=\"${job}\", action=~\"objectPut|objectPutPart|objectCopy|objectPutCopyPart\"}[10m]) * 1000) /\nsum(rate(http_request_duration_seconds_count{namespace=\"${namespace}\", job=\"${job}\", action=~\"objectPut|objectPutPart|objectCopy|objectPutCopyPart\"}[10m]) * 1000)",
```

Comment:
not sure about the `sum` here: the latency is just the `http_request_duration_seconds_sum / http_request_duration_seconds_count` (with `rate` to scope it to 10 minute), making the sum afterwards seems weird...

if needed at all (e.g. not handled by rate) for "merging" the different objectPut/PutPart/...", should we not do the sum() before getting the rate?

Reply:
you can't just use `http_request_duration_seconds_sum / http_request_duration_seconds_count` because both metrics are counter, so they never go down, they always go up. This means the resulting value from the division here will only rise and never go down.

This is why you want to use the `rate()` function, this way you can have a graph that goes up and down regarding if your latency goes up or goes down.

you cannot get a range vector (aka: the syntax `metric{n="toto"}[10m]`) from the resulting `sum` method. you must apply it to your metric, then call functions on the top of it. The `sum` method does not accept range vector so we have only one order of function call: metric -> range vector -> rate -> sum

What can be done is multiplying the result by 1000 after the sum instead of before :thinking: